adapters/mqtt-vertx-base/src/main/java/org/eclipse/hono/adapter/mqtt/CommandSubscription.java

```diff
@@ -159,4 +197,24 @@ public static CommandSubscription fromTopic(final String topic, final Device aut
         }
     }
 
+    /**
+     * Creates a command subscription object for the given topic. When the authenticated device is given
+     * it is used to either check given tenant and device-id from topic or fill this
+     * fields if not given.
+     *
+     * @param topic The topic to subscribe for commands.
+     * @param authenticatedDevice The authenticated device or {@code null}.
+     * @param qos The qos to subscribe for commands.     *
+     * @param clientId The the client identifier as provided by the remote MQTT client.     *
+     * @return The CommandSubscription object or {@code null} if the topic does not match the rules.
+     * @throws NullPointerException if topic is {@code null}.
+     */
+    public static CommandSubscription fromTopic(final String topic, final Device authenticatedDevice, final MqttQoS qos, final String clientId) {
```

Comment:
why don't we simply pass in the `MqttSubscribeMessage` as an argument. It contains all of the other information and `CommandSubscription` is already dependent on vertx-mqtt.
We then only need one factory method, e.g. *fromSubscribe* instead of two, WDYT?

Reply:
In case of subscribe it is ` MqttSubscribeMessage` and in case of unsubscribe it is `MqttUnsubscribeMessage`. IMHO it will be still 2 methods.

Reply:
that's true. However, we might still pass the message itself instead of multiple params which are contained in the message. WDYT?

Reply:
`MqttSubscribeMessage ` and `MqttUnsubscribeMessage` messages contain list of topics. So this cannot be directly passed to _fromSubscribe_ . In case of subscribe, `MqttTopicSubscription` can be sent instead of topic and qos. In case of unsubscribe, it contains list of topic strings, so in this case it remains unchanged.

Reply:
yes, I forgot about that.